--- mongoose.c.orig	2017-02-21 17:49:14.086095834 +0100
+++ mongoose.c	2017-02-21 17:50:13.833186350 +0100
@@ -5359,7 +5359,9 @@
   // won't kill the whole process.
   (void) signal(SIGPIPE, SIG_IGN);
   // Also ignoring SIGCHLD to let the OS to reap zombies properly.
-  (void) signal(SIGCHLD, SIG_IGN);
+  // In cvmfs, we don't want this and we don't let mongoose start
+  // new processes
+  // (void) signal(SIGCHLD, SIG_IGN);
 #endif // !_WIN32
 
   (void) pthread_mutex_init(&ctx->mutex, NULL);
