--- build_detect_platform.orig	2013-04-24 14:58:26.000000000 +0200
+++ build_detect_platform	2013-04-24 14:59:00.000000000 +0200
@@ -146,14 +146,14 @@
 
     # Test whether Snappy library is installed
     # http://code.google.com/p/snappy/
-    $CXX $CFLAGS -x c++ - -o /dev/null 2>/dev/null  <<EOF
-      #include <snappy.h>
-      int main() {}
-EOF
-    if [ "$?" = 0 ]; then
-        COMMON_FLAGS="$COMMON_FLAGS -DSNAPPY"
-        PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -lsnappy"
-    fi
+#    $CXX $CFLAGS -x c++ - -o /dev/null 2>/dev/null  <<EOF
+#      #include <snappy.h>
+#      int main() {}
+#EOF
+#    if [ "$?" = 0 ]; then
+#        COMMON_FLAGS="$COMMON_FLAGS -DSNAPPY"
+#        PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -lsnappy"
+#    fi
 
     # Test whether tcmalloc is available
     $CXX $CFLAGS -x c++ - -o /dev/null -ltcmalloc 2>/dev/null  <<EOF
