--- build_detect_platform.orig	2013-08-15 13:32:42.000000000 +0200
+++ build_detect_platform	2013-08-15 13:34:01.000000000 +0200
@@ -175,14 +175,14 @@
 
     # Test whether Snappy library is installed
     # http://code.google.com/p/snappy/
-    $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT 2>/dev/null  <<EOF
-      #include <snappy.h>
-      int main() {}
-EOF
-    if [ "$?" = 0 ]; then
-        COMMON_FLAGS="$COMMON_FLAGS -DSNAPPY"
-        PLATFORM_LIBS="$PLATFORM_LIBS -lsnappy"
-    fi
+#    $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT 2>/dev/null  <<EOF
+#      #include <snappy.h>
+#      int main() {}
+#EOF
+#    if [ "$?" = 0 ]; then
+#        COMMON_FLAGS="$COMMON_FLAGS -DSNAPPY"
+#        PLATFORM_LIBS="$PLATFORM_LIBS -lsnappy"
+#    fi
 
     # Test whether tcmalloc is available
     $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT -ltcmalloc 2>/dev/null  <<EOF
