--- build_detect_platform.orig	2022-11-21 11:50:01.933197574 +0100
+++ build_detect_platform	2022-11-21 11:50:52.179862263 +0100
@@ -202,12 +202,12 @@
 #    fi
 
     # Test whether tcmalloc is available
-    $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT -ltcmalloc 2>/dev/null  <<EOF
-      int main() {}
-EOF
-    if [ "$?" = 0 ]; then
-        PLATFORM_LIBS="$PLATFORM_LIBS -ltcmalloc"
-    fi
+#    $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT -ltcmalloc 2>/dev/null  <<EOF
+#      int main() {}
+#EOF
+#    if [ "$?" = 0 ]; then
+#        PLATFORM_LIBS="$PLATFORM_LIBS -ltcmalloc"
+#    fi
 
     rm -f $CXXOUTPUT 2>/dev/null
 fi
