--- src/pacparser.c
+++ src/pacparser.c
@@ -154,7 +154,8 @@ resolve_host(const char *hostname, char *ipaddr_list, int max_results,
   error = getaddrinfo(hostname, NULL, &hints, &result);
   if (error) return error;
   int i = 0;
-  for(struct addrinfo *ai = result; ai != NULL && i < max_results; ai = ai->ai_next, i++) {
+  struct addrinfo *ai;
+  for(ai = result; ai != NULL && i < max_results; ai = ai->ai_next, i++) {
     getnameinfo(ai->ai_addr, ai->ai_addrlen, ipaddr, sizeof(ipaddr), NULL, 0,
                 NI_NUMERICHOST);
     if (ipaddr_list[0] == '\0') sprintf(ipaddr_list, "%s", ipaddr);

