--- src/pymod/setup.py.orig	2024-05-10 10:35:55.925174116 +0200
+++ src/pymod/setup.py	2024-05-10 10:36:47.427933459 +0200
@@ -61,20 +61,20 @@
     return ver.replace("-", ".dev")
 
 
-def git_version():
-    return sanitize_version(
-        subprocess.check_output(
-            "git describe --always --tags --candidate=100".split(" "), text=True
-        )
-    )
+#def git_version():
+#    return sanitize_version(
+#        subprocess.check_output(
+#            "git describe --always --tags --candidate=100".split(" "), text=True
+#        )
+#    )
 
 
 def pacparser_version():
-    if (
-        subprocess.call("git rev-parse --git-dir".split(" "), stderr=subprocess.DEVNULL)
-        == 0
-    ):
-        return git_version()
+    #if (
+    #    subprocess.call("git rev-parse --git-dir".split(" "), stderr=subprocess.DEVNULL)
+    #    == 0
+    #):
+    #    return git_version()
 
     # Check if we have version.mk. It's added in the manual release tarball.
     version_file = os.path.join(setup_dir(), "..", "version.mk")

