--- lib/vtls/openssl.c	2016-04-06 14:37:31.000000000 +0200
+++ lib/vtls/openssl.c.patched	2016-04-06 14:37:26.000000000 +0200
@@ -1555,13 +1555,8 @@
     break;
 #endif
   case CURL_SSLVERSION_SSLv3:
-#ifdef USE_TLS_SRP
-    if(data->set.ssl.authtype == CURL_TLSAUTH_SRP)
-      return CURLE_SSL_CONNECT_ERROR;
-#endif
-    req_method = SSLv3_client_method();
-    use_sni(FALSE);
-    break;
+    failf(data, "libcurl was built without SSLv3 support");
+    return CURLE_NOT_BUILT_IN;
   }
 
   if(connssl->ctx)
