diff --git build/linux.inc build/linux.inc
index b2a9abb..b6144ff 100644
--- build/linux.inc
+++ build/linux.inc
@@ -117,16 +117,16 @@ TBB.DEF = $(TBB.LST:.lst=.def)

 TBB.DLL = $(TBB_NO_VERSION.DLL).$(SONAME_SUFFIX)
 TBB.LIB = $(TBB.DLL)
-TBB_NO_VERSION.DLL=libtbb$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
+TBB_NO_VERSION.DLL=libtbb$(CUSTOM_SUFFIX)$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 LINK_TBB.LIB = $(TBB_NO_VERSION.DLL)

-MALLOC_NO_VERSION.DLL = libtbbmalloc$(DEBUG_SUFFIX).$(MALLOC_DLL)
+MALLOC_NO_VERSION.DLL = libtbbmalloc$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(MALLOC_DLL)
 MALLOC.DEF = $(MALLOC_ROOT)/$(def_prefix)-tbbmalloc-export.def
 MALLOC.DLL = $(MALLOC_NO_VERSION.DLL).$(SONAME_SUFFIX)
 MALLOC.LIB = $(MALLOC_NO_VERSION.DLL)
 LINK_MALLOC.LIB = $(MALLOC_NO_VERSION.DLL)

-MALLOCPROXY_NO_VERSION.DLL = libtbbmalloc_proxy$(DEBUG_SUFFIX).$(DLL)
+MALLOCPROXY_NO_VERSION.DLL = libtbbmalloc_proxy$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 MALLOCPROXY.DEF = $(MALLOC_ROOT)/$(def_prefix)-proxy-export.def
 MALLOCPROXY.DLL = $(MALLOCPROXY_NO_VERSION.DLL).$(SONAME_SUFFIX)
 MALLOCPROXY.LIB = $(MALLOCPROXY_NO_VERSION.DLL)
diff --git build/macos.inc build/macos.inc
index b876c42..3f80554 100644
--- build/macos.inc
+++ build/macos.inc
@@ -89,16 +89,16 @@ def_prefix = $(if $(findstring 64,$(arch)),mac64,mac32)

 TBB.LST = $(tbb_root)/src/tbb/$(def_prefix)-tbb-export.lst
 TBB.DEF = $(TBB.LST:.lst=.def)
-TBB.DLL = libtbb$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
+TBB.DLL = libtbb$(CUSTOM_SUFFIX)$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 TBB.LIB = $(TBB.DLL)
 LINK_TBB.LIB = $(TBB.LIB)

 MALLOC.DEF = $(MALLOC_ROOT)/$(def_prefix)-tbbmalloc-export.def
-MALLOC.DLL = libtbbmalloc$(DEBUG_SUFFIX).$(MALLOC_DLL)
+MALLOC.DLL = libtbbmalloc$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(MALLOC_DLL)
 MALLOC.LIB = $(MALLOC.DLL)
 LINK_MALLOC.LIB = $(MALLOC.LIB)

-MALLOCPROXY.DLL = libtbbmalloc_proxy$(DEBUG_SUFFIX).$(MALLOC_DLL)
+MALLOCPROXY.DLL = libtbbmalloc_proxy$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(MALLOC_DLL)
 MALLOCPROXY.LIB = $(MALLOCPROXY.DLL)
 LINK_MALLOCPROXY.LIB = $(MALLOCPROXY.LIB)

