diff --git build/linux.inc build/linux.inc
index 49decc6..5a9b25e 100644
--- build/linux.inc
+++ build/linux.inc
@@ -123,22 +123,22 @@ TBB.DEF = $(TBB.LST:.lst=.def)
 
 TBB.DLL = $(TBB_NO_VERSION.DLL).$(SONAME_SUFFIX)
 TBB.LIB = $(TBB.DLL)
-TBB_NO_VERSION.DLL=libtbb$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
+TBB_NO_VERSION.DLL=libtbb$(CUSTOM_SUFFIX)$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 LINK_TBB.LIB = $(TBB_NO_VERSION.DLL)
 
-MALLOC_NO_VERSION.DLL = libtbbmalloc$(DEBUG_SUFFIX).$(MALLOC_DLL)
+MALLOC_NO_VERSION.DLL = libtbbmalloc$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(MALLOC_DLL)
 MALLOC.DEF = $(MALLOC_ROOT)/$(def_prefix)-tbbmalloc-export.def
 MALLOC.DLL = $(MALLOC_NO_VERSION.DLL).$(SONAME_SUFFIX)
 MALLOC.LIB = $(MALLOC_NO_VERSION.DLL)
 LINK_MALLOC.LIB = $(MALLOC_NO_VERSION.DLL)
 
-MALLOCPROXY_NO_VERSION.DLL = libtbbmalloc_proxy$(DEBUG_SUFFIX).$(DLL)
+MALLOCPROXY_NO_VERSION.DLL = libtbbmalloc_proxy$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 MALLOCPROXY.DEF = $(MALLOC_ROOT)/$(def_prefix)-proxy-export.def
 MALLOCPROXY.DLL = $(MALLOCPROXY_NO_VERSION.DLL).$(SONAME_SUFFIX)
 MALLOCPROXY.LIB = $(MALLOCPROXY_NO_VERSION.DLL)
 LINK_MALLOCPROXY.LIB = $(MALLOCPROXY.LIB)
 
-RML_NO_VERSION.DLL = libirml$(DEBUG_SUFFIX).$(DLL)
+RML_NO_VERSION.DLL = libirml$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 RML.DEF = $(RML_SERVER_ROOT)/lin-rml-export.def
 RML.DLL = $(RML_NO_VERSION.DLL).1
 RML.LIB = $(RML_NO_VERSION.DLL)
diff --git build/macos.inc build/macos.inc
index c020e70..b4bcae4 100644
--- build/macos.inc
+++ build/macos.inc
@@ -105,12 +105,12 @@ def_prefix = $(if $(findstring 64,$(arch)),mac64,mac32)
 
 TBB.LST = $(tbb_root)/src/tbb/$(def_prefix)-tbb-export.lst
 TBB.DEF = $(TBB.LST:.lst=.def)
-TBB.DLL = libtbb$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
+TBB.DLL = libtbb$(CUSTOM_SUFFIX)$(CPF_SUFFIX)$(DEBUG_SUFFIX).$(DLL)
 TBB.LIB = $(TBB.DLL)
 LINK_TBB.LIB = $(TBB.LIB)
 
 MALLOC.DEF = $(MALLOC_ROOT)/$(def_prefix)-tbbmalloc-export.def
-MALLOC.DLL = libtbbmalloc$(DEBUG_SUFFIX).$(MALLOC_DLL)
+MALLOC.DLL = libtbbmalloc$(CUSTOM_SUFFIX)$(DEBUG_SUFFIX).$(MALLOC_DLL)
 MALLOC.LIB = $(MALLOC.DLL)
 LINK_MALLOC.LIB = $(MALLOC.LIB)
 
